nsxt/removed_resource_wrapper.go

```diff
@@ -1,93 +1,86 @@
 package nsxt
 
 import (
+	"context"
 	"log"
 
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 
 	"github.com/vmware/terraform-provider-nsxt/nsxt/util"
 )
 
+const nsxRemovedVersion = "9.0.0"
+
 type resourceFunc func() *schema.Resource
 
 func commonVersionCheck(m interface{}) bool {
 	initNSXVersion(getPolicyConnector(m))
-	return !util.NsxVersionHigherOrEqual("9.0.0")
+	return !util.NsxVersionHigherOrEqual(nsxRemovedVersion)
 }
 
-func readWrapper(originalFunc schema.ReadFunc, name string, fail bool) schema.ReadFunc {
-	wrappedFunc := func(d *schema.ResourceData, m interface{}) error {
+func genericWrapper(originalFunc interface{}, name string, fail bool, action string) interface{} {
+	return func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 		if commonVersionCheck(m) {
-			return originalFunc(d, m)
-
+			switch f := originalFunc.(type) {
```

Comment:
Thanks for this. I could not get this cast to work properly, and this is why I did not implement a generic wrapper in the first place.